apollo-router/src/query_planner/mod.rs

```diff
@@ -167,6 +167,22 @@ impl PlanNode {
             let mut value;
             let mut errors;
 
+            if !current_dir.is_empty() {
+                // Check if parent data has field requested by Flatten
+                match parent_value.pointer(&current_dir.to_string()) {
```

Comment:
it's not the right place to do that. The check should be performed at variables aggregation in https://github.com/apollographql/router/blob/main/apollo-router/src/query_planner/mod.rs#L492-L508
as you can see there's already a check to avoid a subquery if the list of variables is empty, this needs an additional check that the variables are not null

Reply:
Thanks for feedback!
Do I need to perform this check inside `match`?

Reply:
Also, the problem is that with my example, variables are empty object (`{}`) in both cases

Reply:
@Geal, fixed, hope I understood you right 🙂

Reply:
@Geal, fixed, hope I understood you right. 🙂

Reply:
sorry :sweat: I was not very clear, I think it has to be done inside `Variables::new`. If you want I can take a closer look at it and add some tests

Reply:
I have free time, so I can do it by myself 🙂
One question: do I need to add this check [inside this](https://github.com/apollographql/router/blob/main/apollo-router/src/query_planner/mod.rs#L457) `else` branch?

Reply:
@Geal, check latest commit please 🙂

Reply:
thanks, looking into it